cl/corpus_importer/management/commands/harvard_merge.py

```diff
@@ -0,0 +1,895 @@
+    else:
+        # Only save attorneys
+        find_attorneys_fields = soup.find_all(
+            lambda tag: tag.get("data-type") == "attorneys"
+            or (tag.name == "attorneys" and tag.get("data-type") is None)
+        )
+        if find_attorneys_fields:
+            attorneys = " ".join(str(x) for x in find_attorneys_fields)
+            all_data["attorneys"] = attorneys
+
+    if "otherdate" in short_data:
+        # Rename to correct field name
+        short_data["other_dates"] = short_data.pop("otherdate")
+
+    if "seealso" in short_data:
+        # Rename to correct field name
+        short_data["cross_reference"] = short_data.pop("seealso")
+
+    long_data = parse_extra_fields(soup, long_fields, True)
+    all_data.update(short_data)
+    all_data.update(long_data)
+    all_data = {k: v for k, v in all_data.items() if v}
+    return all_data
+
+
+def combine_non_overlapping_data(
+    cluster_id: int, harvard_data: dict
+) -> dict[str, Tuple]:
+    """Combine non overlapping data and return dictionary of data for merging
+
+    :param cluster_id: Cluster id to merge
+    :param harvard_data: The harvard data as json
+    :return: Optional dictionary of data to continue to merge
+    """
+    opinion_cluster = OpinionCluster.objects.get(id=cluster_id)
+    all_data = fetch_non_harvard_data(harvard_data)
+    changed_values_dictionary: dict[str, Tuple] = {}
+    for key, value in all_data.items():
+        cl_value = getattr(opinion_cluster, key)
+        if not cl_value:
+            # Value is empty for key, we can add it directly to the object
+            OpinionCluster.objects.filter(id=cluster_id).update(**{key: value})
+        else:
+            if value != cl_value:
+                # We have different values, update dict
+                changed_values_dictionary[key] = (value, cl_value)
+
+    return changed_values_dictionary
+
+
+def merge_long_fields(
+    cluster_id: int,
+    field_name: str,
+    overlapping_data: Optional[Tuple[str, str]],
+) -> None:
+    """Merge two long text fields
+
+    :param cluster_id: Cluster id to update
+    :param field_name: field name to update in opinion cluster
+    :param overlapping_data: data to compare from harvard and courtlistener
+    :return: None
+    """
+    if overlapping_data:
+        if overlapping_data[0] and overlapping_data[1]:
+            harvard_data, cl_data = overlapping_data[0], overlapping_data[1]
+            # Do some text comparison
+            similarity = get_cosine_similarity(harvard_data, cl_data)
+            if similarity < 0.9:
+                # they are not too similar, choose the larger one
+                if len(harvard_data) > len(cl_data):
+                    OpinionCluster.objects.filter(id=cluster_id).update(
+                        **{field_name: harvard_data}
+                    )
+            else:
+                pass
+                # should we log long data not really being similar?
```

Comment:
I think so. You get the chance to log only once.

Reply:
Yeah - we should log if it's less than 0.5 or some threshold and review them.